pandas/core/generic.py

```diff
@@ -3357,6 +3361,12 @@ def __setattr__(self, name, value):
                 else:
                     object.__setattr__(self, name, value)
             except (AttributeError, TypeError):
+                if isinstance(self, ABCDataFrame) and (is_list_like(value)):
+                    warnings.warn("Pandas doesn't allow Series to be assigned "
+                                  "into nonexistent columns - see "
```

Comment:
I would write this a bit more generic, as it does not only raise for Series objects (and I also find it a bit confusing, as pandas certainly allows assigning Series objects, just with another syntax). 

Maybe something like "pandas doesn't allow to add a new column using attribute access" ? (can certainly be improved further)